presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java

```diff
@@ -210,6 +210,7 @@ public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Optional
             allSymbols.addAll(groupBySymbols);
             allSymbols.addAll(nonDistinctAggregateSymbols);
             allSymbols.add(distinctSymbol);
+            allSymbols = allSymbols.stream().distinct().collect(Collectors.toList());
```

Comment:
@martint Do you mean use `Set<Symbol>` instead of `List<Symbol>` for `allSymbols` and `groupByKeys` ? :)